com.unity.ml-agents/Runtime/InferenceBrain/ModelRunner.cs

```diff
@@ -130,7 +127,15 @@ public void PutObservations(AgentInfo info, List<ISensor> sensors, Action<AgentA
                 sensors = sensors
             });
 
-            m_ActionFuncs.Add(new AgentIdActionPair { action = action, agentId = info.episodeId });
+            m_ActionId.Add(info.episodeId);
```

Comment:
Add a note that we always need to append the ID (even if the agent is done) in order to preserver the ordering.